src/compiler/checker.ts

```diff
@@ -12100,6 +12121,7 @@ namespace ts {
 
                 if (node.body.kind === SyntaxKind.Block) {
                     checkSourceElement(node.body);
+                    checkUnusedLocals(node);

```

Comment:
i think you just need to call checkUnusedParameters after checkExpression in the else clause.